### DISCUSSION TITLE
Create a temperature field and heat conduction

### DISCUSSION THREAD

**Discussion by wenpeng231024** [idaholab/moose#16234] [Q&A General] [answered]
Hi everyone,

I want to create a temperature field (the central region is hot).  I use an AuxVariables named "T" and FunctionAux of Auxkernel to create this field. It can work. But when I use "HeatConduction" kernel to couple the heat conduction equation, there is the error "No nonlinear variable named T found. Did you specify an auxiliary variable when you meant to specify a nonlinear variable (or vice-versa)?"

[AuxVariables]
  [T]
  []
[]
[AuxKernels]
    [T_bounds]
    type = FunctionAux 
    variable = T
    function = '690+0.25*(710-690)*(cos(x/9-1)*3.1415926+1)*(cos(y/9-1)*3.1415926+1)'
  []
[]

 [heat]         
    type = HeatConduction
    variable = T
  []

My question is how can I build this temperature field and use the "HeatConduction" kernel at the same time? 
 
Best,

Peng

**Comment by WilkAndy** [comment, marked as answer]
Hey Peng,

The problem here is that when you're using `T` in the `HeatConduction` Kernel, you are telling MOOSE that you want the solution for `T` assuming it is governed by the heat equation.  However, when you set `T` to be an AuxVariable, you are telling MOOSE that you want to control the dynamics of `T` using the `FunctionAux`. 

You've got to decide which one you want to do.   Is `T` a dynamical variable, or are you setting `T` externally using your `FunctionAux`?

Perhaps what you're trying to do is set an initial condition for `T`, and then solve for its evolution?  If so, use a `FunctionIC`.

**Reply by wenpeng231024** [reply]
Hi WilkAndy,

Thanks for your response. Actually, I want to set this temperature field (hot center and cold boundary) as the boundary conditions. 

for example, if I set the left boundary temperature is 600 K and the right boundary temperature is 700 K. There is a temperature gradient in the domain. Then I couple the heat conduction equation to research the effect of void or bubble on the temperature gradient due to its different thermal conductivity.

But I found I just can set the top, bottom, left, and right for the boundary. So I don't know how to create this temperature field (hot center and cold boundary) using these boundaries. If I use FuntionIC setting this temperature field as the initial condition, I found the temperature will quickly change to a constant for the whole system. 

So, what I want to do is create a stable temperature field (hot center and cold boundary) like boundary condition. And couple the heat conduction equation to research the effect of void or bubble on the temperature gradient. Do you know how to do this?

Best,

Peng

**Reply by WilkAndy** [reply]
Perhaps your thermal conductivity was too high?  If you set it very small then the temperature will only change very slowly.

**Reply by wenpeng231024** [reply]
Hi Andy,

Yes, I agree with you. Maybe I need to use a smaller temperature gradient or lower thermal conductivity. I will try these.

Thanks for your help.

Peng
